src/node/internal/crypto_sign.ts

```diff
@@ -0,0 +1,444 @@
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/* TODO: the following is adopted code, enabling linting one day */
+/* eslint-disable */
+
+import {
+  default as cryptoImpl,
+  type SignHandle,
+  type VerifyHandle,
+  type CreateAsymmetricKeyOptions,
+} from 'node-internal:crypto';
+
+import { validateString } from 'node-internal:validators';
+
+import { Buffer } from 'node-internal:internal_buffer';
+
+import {
+  KeyObject,
+  isKeyObject,
+  getKeyObjectHandle,
+  createPrivateKey,
+  createPublicKey,
+} from 'node-internal:crypto_keys';
+
+import {
+  ERR_CRYPTO_SIGN_KEY_REQUIRED,
+  ERR_INVALID_ARG_TYPE,
+  ERR_INVALID_ARG_VALUE,
+} from 'node-internal:internal_errors';
+
+import { Writable } from 'node-internal:streams_writable';
+import { isArrayBufferView } from 'node-internal:internal_types';
+
+export interface SignOptions {}
+
+export interface Sign extends Writable {
+  [kHandle]: SignHandle;
+  update(data: any): void;
+  sign(privateKey: any): any;
+}
+export interface Verify extends Writable {
+  [kHandle]: VerifyHandle;
+  update(data: any): void;
+  verify(publicKey: any, signature: any): any;
+}
+
+const kHandle = Symbol('kHandle');
+
+// Uses old-style class syntax for Node.js compatibility
+export const Sign = function (this: Sign, algorithm: string, options: any) {
+  if (!(this instanceof Sign)) return new Sign(algorithm, options);
+  validateString(algorithm, 'algorithm');
+  this[kHandle] = new cryptoImpl.SignHandle(algorithm);
+  Writable.call(this, options);
+  return this;
+} as any as {
+  new (algorithm: string, options?: SignOptions): Sign;
+};
+Object.setPrototypeOf(Sign.prototype, Writable.prototype);
+Object.setPrototypeOf(Sign, Writable);
+
+Sign.prototype._write = function _write(
+  chunk: string | ArrayBufferView,
+  encoding: string | undefined | null,
+  callback: (err?: any) => void
+) {
+  this.update(chunk, encoding);
+  callback();
+};
+
+Sign.prototype.update = function (
+  this: Sign,
+  data: string | ArrayBufferView,
+  encoding?: string
+) {
+  if (typeof data === 'string') {
+    data = Buffer.from(data, encoding);
+  }
+  if (!isArrayBufferView(data)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'data',
+      ['string', 'Buffer', 'TypedArray', 'DataView'],
+      data
+    );
+  }
+  this[kHandle].update(data);
+  return this;
+};
+
+function getIntOption(name: string, options: any): number | undefined {
```

Comment:
```suggestion
function getIntOption(name: string, options: Record<string, unknown>): number | undefined {
```

Reply:
It's not a `Record<string, unknown>`. It might be a `KeyObject`, and `CryptoKey`, etc.